src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -533,6 +613,17 @@ struct function3_get : QueryRequestType
   }
 };
 
+template <typename QueryRequestType, typename Getter>
+struct function4_get : virtual QueryRequestType
+{
+  boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const
+  {
+    auto k = QueryRequestType::key;
+    return Getter::get(device, k, arg1);
```

Comment:
Just to point out what may look  inconsistent but in fact is not:  template types and always CamelCaseTypes by convention :-)